src/scenekit.cs

```diff
@@ -443,9 +467,196 @@ interface SCNCamera : SCNAnimatable, SCNTechniqueSupport, NSCopying, NSSecureCod
+	
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[Export ("focusDistance")]
+		nfloat FocusDistance { get; set; }
+	
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[Export ("focalBlurSampleCount")]
+		nint FocalBlurSampleCount { get; set; }
+
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[Export ("fStop")]
+		nfloat FStop { get; set; }
+	
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[Export ("apertureBladeCount")]
+		nint ApertureBladeCount { get; set; }
+	
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[Export ("screenSpaceAmbientOcclusionIntensity")]
+		nfloat ScreenSpaceAmbientOcclusionIntensity { get; set; }
+	
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[Export ("screenSpaceAmbientOcclusionRadius")]
+		nfloat ScreenSpaceAmbientOcclusionRadius { get; set; }
+	
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[Export ("screenSpaceAmbientOcclusionBias")]
+		nfloat ScreenSpaceAmbientOcclusionBias { get; set; }
+	
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[Export ("screenSpaceAmbientOcclusionDepthThreshold")]
+		nfloat ScreenSpaceAmbientOcclusionDepthThreshold { get; set; }
+	
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[Export ("screenSpaceAmbientOcclusionNormalThreshold")]
+		nfloat ScreenSpaceAmbientOcclusionNormalThreshold { get; set; }
+
 #endif
 	}
 
+	[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+	[Protocol]
+	[BaseType (typeof(NSObject))]
+	interface SCNCameraControlConfiguration
+	{
+		[Abstract]
+		[Export ("autoSwitchToFreeCamera")]
+		bool AutoSwitchToFreeCamera { get; set; }
+	
+		[Abstract]
+		[Export ("allowsTranslation")]
+		bool AllowsTranslation { get; set; }
+	
+		[Abstract]
+		[Export ("flyModeVelocity")]
+		nfloat FlyModeVelocity { get; set; }
+	
+		[Abstract]
+		[Export ("panSensitivity")]
+		nfloat PanSensitivity { get; set; }
+	
+		[Abstract]
+		[Export ("truckSensitivity")]
+		nfloat TruckSensitivity { get; set; }
+	
+		[Abstract]
+		[Export ("rotationSensitivity")]
+		nfloat RotationSensitivity { get; set; }
+	}
+
+	interface ISCNCameraControllerDelegate {}
+	
+	[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+	[Protocol]
+	[Model] // Figured I would keep the model for convenience, as all the methods here are optional
+	[BaseType (typeof(NSObject))]
+	interface SCNCameraControllerDelegate
+	{
+		[Export ("cameraInertiaWillStartForController:")]
+		void CameraInertiaWillStart (SCNCameraController cameraController);
+	
+		[Export ("cameraInertiaDidEndForController:")]
+		void CameraInertiaDidEnd (SCNCameraController cameraController);
+	}
+	
+	[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+	[BaseType (typeof(NSObject))]
+	interface SCNCameraController
+	{
+		[Wrap ("WeakDelegate")]
+		[NullAllowed]
+		[Protocolize]
+		SCNCameraControllerDelegate Delegate { get; set; }
+	
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Assign)]
+		NSObject WeakDelegate { get; set; }
```

Comment:
That scenario is still doable with the interfaces, you would end up with one (or few) extra member(s) without `[Export]` and the IDE would help/guide you to create the members. The main advantage of interfaces is that can be used in other API (from us or 3rd parties) while a custom `NSObject`won't be (forcing the use of the above).

OTOH it's easy to support that extra scenario, it's not a breaking change to add them and the code saving (of not having then) are not huge. It will be scheduled among the post-`xcode9` work.